theme/views.py

```diff
@@ -614,34 +616,31 @@ def create_irods_account(request):
                                      uname=settings.LINUX_ADMIN_USER_FOR_HS_USER_ZONE,
                                      pwd=settings.LINUX_ADMIN_USER_PWD_FOR_HS_USER_ZONE,
                                      exec_cmd=exec_cmd)
-            if output:
-                if 'ERROR:' in output.upper() and \
-                        not 'CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME' in output.upper():
+            for out_str in output:
+                if 'bash:' in out_str or ('ERROR:' in out_str.upper() and \
+                        not 'CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME' in out_str.upper()):
                     # there is an error from icommand run which is not about the fact 
                     # that the user already exists, report the error
-                    return HttpResponse(
-                        dumps({"error": 'iRODS server failed to create this iRODS account {0}. '
-                                        'Check the server log for details.'.format(user.username)}),
-                        content_type = "application/json"
+                    return JsonResponse(
+                        {"error": 'iRODS server failed to create this iRODS account {0}. '
+                                  'Check the server log for details.'.format(user.username)},
+                        status=status.HTTP_500_INTERNAL_SERVER_ERROR
                     )
 
             user_profile = UserProfile.objects.filter(user=user).first()
             user_profile.create_irods_user_account = True
             user_profile.save()
-            return HttpResponse(
-                    dumps({"success": "iRODS account {0} is created successfully".format(
-                        user.username)}),
-                    content_type = "application/json"
+            return JsonResponse(
+                    {"success": "iRODS account {0} is created successfully".format(user.username)},
+                    status=status.HTTP_200_OK
             )
         except Exception as ex:
-            return HttpResponse(
-                    dumps({"error": ex.message + ' - iRODS server failed to create this '
-                                                 'iRODS account.'}),
-                    content_type = "application/json"
+            return JsonResponse(
+                    {"error": ex.message + ' - iRODS server failed to create this iRODS account.'},
+                    status=status.HTTP_500_INTERNAL_SERVER_ERROR
             )
     else:
```

Comment:
You should be able to remove the check for POST and this response if you use the annotation `@api_view(['POST'])`

Reply:
This view function does not have a REST API endpoint, so I don't think I should add ```@api_view('POST'])``` decorator at this point.

Reply:
I don't see why it couldn't be a part of hsapi.  Even then, if you wanted to exclude it from the list endpoints in swagger then you can use the annotation `@swagger_auto_schema(auto_schema=None)`.  I feel like using annotations and letting existing libraries do the heavy lifting where possible is easier to read and support than writing our own.

Reply:
@sblack-usu Definitely agreed regarding your point "I feel like using annotations and letting existing libraries do the heavy lifting where possible is easier to read and support than writing our own." I do think it would work better if we address this when needed, i.e., adding api_view decorator when we want to expose this as a REST API as opposed to adding the decorator in advance, but if you think we should do it now, I can add it in as part of this PR as well. Just let me know.